xlators/features/shard/src/shard.c

```diff
@@ -6939,6 +7005,35 @@ shard_common_inode_write_begin(call_frame_t *frame, xlator_t *this,
     uint64_t block_size = 0;
     shard_local_t *local = NULL;
 
+    if (frame->root->pid == GF_CLIENT_PID_GSYNCD) {
```

Comment:
We also have this switch handling after getting the block size. To avoid code duplication, we can have an inline function for this